src/__tests__/branchIfFalsy.tsx

```diff
@@ -0,0 +1,115 @@
+
+describe('branchIfFalsy()', () => {
+  test('abort rendering when prop is falsy', () => {
+    interface Props {
+      x?: number | null
+      testId: string
+    }
+
+    const Component: FC<Props> = flowMax(
+      branchIfFalsy('x'),
+      addProps(({x}) => ({
+        x: x + 3,
+      })),
+      ({x, testId}) => <div data-testid={testId}>{x}</div>,
+    )
+
+    const testId = 'branch-if-falsy'
+    const {rerender} = render(<Component testId={testId} />)
+    expect(screen.queryByTestId(testId)).toBeNull()
+    rerender(<Component x={2} testId={testId} />)
+    expect(screen.getByTestId(testId)).toHaveTextContent('5')
+    rerender(<Component x={null} testId={testId} />)
+    expect(screen.queryByTestId(testId)).toBeNull()
+    rerender(<Component x={undefined} testId={testId} />)
+    expect(screen.queryByTestId(testId)).toBeNull()
+    rerender(<Component x={0} testId={testId} />)
+    expect(screen.queryByTestId(testId)).toBeNull()
+  })
+  test('accepts returns option', () => {
+    interface Props {
+      x?: number | null
+      testId: string
+    }
+
+    const Component: FC<Props> = flowMax(
+      branchIfFalsy('x', {
+        returns: ({testId}) => <div data-testid={testId}>aborted</div>,
+      }),
+      addProps(({x}) => ({
+        x: x + 3,
+      })),
+      ({x, testId}) => <div data-testid={testId}>{x}</div>,
+    )
+
+    const testId = 'branch-if-falsy-returns'
+    const {rerender} = render(<Component testId={testId} />)
+    expect(screen.getByTestId(testId)).toHaveTextContent('aborted')
+    rerender(<Component x={2} testId={testId} />)
+    expect(screen.getByTestId(testId)).toHaveTextContent('5')
+    rerender(<Component x={null} testId={testId} />)
+    expect(screen.getByTestId(testId)).toHaveTextContent('aborted')
+    rerender(<Component x={undefined} testId={testId} />)
+    expect(screen.getByTestId(testId)).toHaveTextContent('aborted')
+  })
+  test('accepts array of prop names', () => {
+    interface Props {
+      x?: number | null
+      testId: string
+    }
+
+    const Component: FC<Props> = flowMax(
+      branchIfFalsy(['x']),
+      addProps(({x}) => ({
+        x: x + 3,
+      })),
+      ({x, testId}) => <div data-testid={testId}>{x}</div>,
+    )
+
+    const testId = 'branch-if-falsy-array'
+    const {rerender} = render(<Component testId={testId} />)
+    expect(screen.queryByTestId(testId)).toBeNull()
+    rerender(<Component x={2} testId={testId} />)
+    expect(screen.getByTestId(testId)).toHaveTextContent('5')
+    rerender(<Component x={null} testId={testId} />)
+    expect(screen.queryByTestId(testId)).toBeNull()
+    rerender(<Component x={undefined} testId={testId} />)
+    expect(screen.queryByTestId(testId)).toBeNull()
+  })
+  test('type-narrows boolean prop', () => {
+    interface Props {
+      x?: boolean | null
+      testId: string
+    }
+
+    const Component: FC<Props> = flowMax(
+      branchIfFalsy('x'),
+      addProps(({x}) => ({
+        y: x && 'abc',
```

Comment:
For `branchIfFalsy()`, can successfully narrow a boolean prop type to just `true` (non-nullish and not `false`)